### DISCUSSION TITLE
How to use RISCV RVV1.0 in FS?

### DISCUSSION THREAD

**Discussion by AlutionXian** [gem5/gem5#1159] [Q&A]
Hi all: 
In light of the release notes for version 23.1.0.0, I've learned that gem5 now natively supports RVV 1.0 in FS mode, which fits perfectly with a project I'm currently working on. I came across a previous discussion thread which mentioned that enabling RVV in FS mode would require changes to the gem5 codebase and the use of OpenSBI, but no further details were given. I would like to know how I can use rvv1.0 in FS mode , a reference document or other more detailed guidance would be very helpful.

**Comment by hnpl** [comment]
Hello there. The gem5 change I referred to here (https://github.com/orgs/gem5/discussions/523#discussioncomment-9658273) has been merged to gem5. [That change](https://github.com/gem5/gem5/pull/390) enabled support for using OpenSBI as a firmware/bootloader.

You can start from this configuration, https://github.com/gem5/gem5/blob/v23.1.0.0/configs/example/gem5_library/riscv-ubuntu-run.py, to use gem5 with the new changes. The configuration will download relevant artifacts (the disk image, the bootloader, the kernel as detailed here https://resources.gem5.org/resources/riscv-ubuntu-20.04-boot?version=3.0.0), and boots up Ubuntu 20.04, and then exits the simulation.

You can modify the disk image to add binaries compiled with RVV instructions (e.g., by cross-compiling RVV binaries on your host then mounting the disk to your host, and then copying the RVV binaries to the mounted disk). After this step, you should be able to run the RVV binaries in your guest system. To run the simulation with a customized disk image, or a modified Linux/OpenSBI version, you can change the input to the `board.set_workload()` function in the example configuration.

**Reply by AlutionXian** [reply]
Thank you for your guidance, which is very helpful to me.
